File: AddBinary/AddBinary.py
class Solution(object):
    def addBinary(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: str
        """
        m = 0
        i = len(a) - 1
        j = len(b) - 1
        c = ''
        while i >= 0 or j >= 0 or m == 1:
            if i >= 0:
                m += int(a[i])
                i -= 1
            else:
                m += 0
            if j >= 0:
                m += int(b[j])
                j -= 1
            else:
                m += 0
            c = str(m % 2) + c
            m /= 2
        return c

File: AddBinary/AddBinary.cpp
#include <string>
#include <iostream>

using namespace std;

class Solution {
public:
	string addBinary(string a, string b) {
		int m = 0;
		int i = a.size() - 1;
		int j = b.size() - 1;
		string c = "";
		while (i >= 0 || j >= 0 || m == 1) {
			if (i >= 0) {
				m += a[i] - '0';
				i -= 1;
			}
			else
				m += 0;
			if (j >= 0) {
				m += b[j] - '0';
				j -= 1;
			}
			else
				m += 0;
			c = to_string(m % 2) + c;
			m /= 2;
		}
		return c;
	}
};